hail/python/hailtop/batch/backend.py

```diff
@@ -445,12 +445,15 @@ def __init__(self,
         gcs_kwargs = {'project': google_project}
         self.__fs: AsyncFS = RouterAsyncFS(default_scheme='file', gcs_kwargs=gcs_kwargs)
 
+        self._batch_client: Optional[BatchClient] = None
```

Comment:
This seems wrong, we set `self._batch_client` earlier in the constructor. Could you explain why this fixes the issue?